src/czbenchmarks/file_utils.py

```diff
@@ -89,13 +90,20 @@ def download_file_from_remote(
         - If the file already exists in the cache and is not expired, it will not be downloaded again.
         - Unsigned requests are tried first, followed by signed requests if the former fails.
     """
+
     cache_manager = (
         _default_cache_manager if cache_dir is None else CacheManager(cache_dir)
     )
-    try:
+
+    protocol = urlparse(remote_url).scheme
```

Comment:
Reviewer note: The modfications to `download_file_from_remote` are not specifically required for PR, but it seemed the error messages could be improved for this function. And there was no test coverage -- tests have been added.